File: ts/project/.sdk/src/cmp/js/fragment/Config.fragment.js

const { BaseFeature } = require('./feature/base/BaseFeature')
// #ImportFeatures


const FEATURE_CLASS = {
  // #FeatureClasses
}


class Config {

  makeFeature(fn) {
    const fc = FEATURE_CLASS[fn]
    const fi = new fc()
    // TODO: errors etc
    return fi
  }

  // False for a feature added at runtime via options.extend (station's
  // adopt path) - the constructor uses this to skip makeFeature for names
  // no generated class backs.
  hasFeature(fn) {
    return null != FEATURE_CLASS[fn]
  }


  main = {
    name: '$$const.Name$$',
    // #MainMeta
  }


  feature = {
    // #FeatureConfigs
  }


  options = {
    base: 'BASEURL',

    'SERVERBLOCK''AUTHBLOCK'headers: 'HEADERS',

    entity: {
      // #EntityConfigs
    }
  }


  entity = 'ENTITYMAP'
}


const config = new Config()

module.exports = {
  config
}

File: ts/project/.sdk/src/cmp/dart/fragment/Config.fragment.dart
import 'feature/base/BaseFeature.dart';
// #ImportFeatures

// ignore: non_constant_identifier_names
final Map<String, BaseFeature Function()> FEATURE_CLASS = {
  // #FeatureClasses
};

class Config {
  BaseFeature makeFeature(String fn) {
    final fc = FEATURE_CLASS[fn];
    if (null == fc) {
      // TODO: errors etc
      throw StateError('Unknown feature: ' + fn);
    }
    return fc();
  }

  final Map<String, dynamic> main = <String, dynamic>{
    'name': 'ProjectName',
  };

  final Map<String, dynamic> feature = <String, dynamic>{
    // #FeatureConfigs
  };

  final Map<String, dynamic> options = <String, dynamic>{
    'base': '$$main.kit.info.servers.0.url$$',

    'AUTHBLOCK''headers': 'HEADERS',

    'entity': <String, dynamic>{
      // #EntityConfigs
    }
  };

  final Map<String, dynamic> entity = 'ENTITYMAP';

  // The pipeline context carries the config as a plain map.
  Map<String, dynamic> toMap() => <String, dynamic>{
        'main': main,
        'feature': feature,
        'options': options,
        'entity': entity,
      };
}

final config = Config();
